spyder/widgets/sourcecode/base.py

```diff
@@ -34,7 +33,8 @@
 from spyder.widgets.calltip import CallTipWidget
 from spyder.widgets.mixins import BaseEditMixin
 from spyder.widgets.sourcecode.terminal import ANSIEscapeCodeHandler
-
+from spyder.widgets.sourcecode.api.decoration import TextDecoration
+from spyder.widgets.sourcecode.utils.decoration import TextDecorationsManager

```

Comment:
Why we have the manager outside of the api, is because we don't need/want to expose that?

Reply:
Hmm yes, I don't think there is a reason to expose it

Normally you will create new TextDecoration and append them to the editor.decorations.